src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -670,6 +670,32 @@ private void FlushPendingOutput()
                         }
                     }
 
+                    //create the file in the main thread and flush the contents in the background thread
+                    // In case , there is an error in file creation, the logic would return from here and
+                    // not consume the exception
+                    try
+                    {
+                        string baseDirectory = Path.GetDirectoryName(transcript.Path);
+                        if(baseDirectory.Equals(transcript.Path.TrimEnd(System.IO.Path.DirectorySeparatorChar)))
+                        {
+                            Exception exception = PSTraceSource.NewArgumentException(transcript.Path,
+                            InternalHostUserInterfaceStrings.InvalidTranscriptFilePath, transcript.Path);
+                            throw exception;
+                        }
```

Comment:
How about `Path.GetFileName(path)`? If the returned file name is null or empty then there is no file specified in the path, which is what this if check is doing.